### PR TITLE
feat: add rule aria-deprecated-role

### PR DESCRIPTION
**Pull request by straker**
Split out `aria-deprecated-role` from `aria-roles` into it's own rule.

Closes: #4069

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -144,6 +144,7 @@
     "html-entities": "1.x",
     "http-server": "^14.1.1",
     "husky": "^8.0.3",
+    "inquirer": "^8.2.5",
```
Comment (straker) [inline package.json:147]: This was needed to run the run generator. Not sure when it was removed.

File: lib/rules/aria-deprecated-role.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "id": "aria-deprecated-role",
+  "selector": "[role]",
+  "matches": "no-empty-role-matches",
+  "tags": ["cat.aria", "wcag2a", "wcag412"],
```
Comment (WilcoFiers) [inline lib/rules/aria-deprecated-role.json:5]: Add the EN tags

File: test/integration/rules/aria-roles/aria-roles.html [hunk 3]
```diff
@@ -131,10 +131,6 @@
   <!-- invalid roles -->
   <div role="lol" id="fail13">fail</div>
   <!-- unsupported roles -->
-  <!-- deprecated roles-->
-  <div role="doc-biblioentry" id="fail14">fail</div>
-  <div role="doc-endnote" id="fail15">fail</div>
-  <div role="directory" id="fail16">ok</div>
```
Comment (WilcoFiers) [inline test/integration/rules/aria-roles/aria-roles.html:137]: Maybe move these to passed?

File: test/integration/rules/aria-deprecated-role/aria-deprecated-role.html [hunk 4]
```diff
@@ -0,0 +1,4 @@
+<!-- deprecated roles-->
```
Comment (WilcoFiers) [inline test/integration/rules/aria-deprecated-role/aria-deprecated-role.html:1]: Probably add some passed examples.

**Review by WilcoFiers — DISMISSED** [review]

**Review by WilcoFiers — APPROVED** [review]

File: lib/rules/aria-deprecated-role.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
+{
+  "id": "aria-deprecated-role",
+  "selector": "[role]",
+  "matches": "no-empty-role-matches",
+  "tags": ["cat.aria", "wcag2a", "wcag412", "EN-301-549", "EN-9.4.1.2"],
+  "impact": "minor",
+  "actIds": ["674b10"],
+  "metadata": {
+    "description": "Ensures elements do not use deprecated roles",
+    "help": "ARIA roles used must not be deprecated"
```
Comment (WilcoFiers) [inline lib/rules/aria-deprecated-role.json:10]: I think you mean this:

```suggestion
    "help": "Deprecated ARIA roles must not be used"
```

**Review by WilcoFiers — CHANGES_REQUESTED** [review]
Nitpick

File: locales/_template.json [hunk 6]
```diff
@@ -21,6 +21,10 @@
       "description": "Ensures every ARIA button, link and menuitem has an accessible name",
       "help": "ARIA commands must have an accessible name"
     },
+    "aria-deprecated-role": {
+      "description": "Ensures elements do not use deprecated roles",
+      "help": "ARIA roles used must not be deprecated"
```
Comment (WilcoFiers) [inline locales/_template.json:26]: I don't think you reran the build script?

**Review by WilcoFiers — CHANGES_REQUESTED** [review]

**Review by WilcoFiers — APPROVED** [review]

**Pull request merged**